eZ/Bundle/EzPublishElasticsearchBundle/DependencyInjection/Compiler/ConnectionParameterPass.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * This file is part of the eZ Publish Kernel package
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+
+namespace eZ\Bundle\EzPublishElasticsearchBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Base class for Compiler passes using ConnectionParamFactory.
+ *
+ * Compiler passes extending this one will usually replace single argument of
+ * a specific service with a parameter of a search engine connection, resolved for a
+ * current siteaccess.
+ */
+abstract class ConnectionParameterPass implements CompilerPassInterface
+{
+    /**
+     * ConnectionParameterFactory service container id.
+     *
+     * @see \eZ\Bundle\EzPublishElasticsearchBundle\ApiLoader\ConnectionParameterFactory
+     *
+     * @var string
+     */
+    protected $factoryId = "ezpublish.elasticsearch.connection_parameter_factory";
```

Comment:
Changed to constructor injection in https://github.com/ezsystems/ezpublish-kernel/commit/7044dc3f754205de372942b724fa3003cc144db5.